### ISSUE TITLE
Variable type unknown when returning caught exception

### ISSUE THREAD

**Issue by layday** [microsoft/pyright#849]
**Describe the bug**
When matching an instance of an exception in a try-except and returning it
the return type is reported to by unknown:

```json
{
	"resource": "test.py",
	"owner": "_generated_diagnostic_collection_name_#0",
	"severity": 8,
	"message": "Return type is unknown",
	"source": "Pyright (reportUnknownVariableType)",
	"startLineNumber": 7,
	"startColumn": 16,
	"endLineNumber": 7,
	"endColumn": 19
}
```

**To Reproduce**
Type check the snippet below.

**Expected behavior**
That the return type would be `Exception`.

**Screenshots or Code**

```py3
# pyright: strict

def foo() -> Exception:
    try:
        ...
    except Exception as exc:
        return exc
    return Exception()
```

**VS Code extension or command-line**
extension, 1.1.39

**Comment by layday** [comment]
I've just noticed my extension is woefully out of date.  It's not been automatically updating.

**Comment by erictraut** [comment]
Yeah, this is working for me with the latest version.

**Comment by layday** [comment]
Thank you, this does work with the latest version.  Closing.
